core/frontend/src/render/webgl/Target.ts

```diff
@@ -739,32 +745,70 @@ export abstract class Target extends RenderTarget implements RenderTargetDebugCo
     // We can't reuse the previous frame's data for a variety of reasons, chief among them that some types of geometry (surfaces, translucent stuff) don't write
     // to the pick buffers and others we don't want - such as non-pickable decorations - do.
     // Render to an offscreen buffer so that we don't destroy the current color buffer.
-    const texture = TextureHandle.createForAttachment(rect.width, rect.height, GL.Texture.Format.Rgba, GL.Texture.DataType.UnsignedByte);
-    if (undefined === texture) {
+    const resources = this.createOrReuseReadPixelResources(rect);
+    if (resources === undefined) {
       receiver(undefined);
       return;
     }
 
     let result: Pixel.Buffer | undefined;
-    const fbo = FrameBuffer.create([texture]);
-    if (undefined !== fbo) {
-      this.renderSystem.frameBufferStack.execute(fbo, true, false, () => {
-        this._drawNonLocatable = !excludeNonLocatable;
-        result = this.readPixelsFromFbo(rect, selector);
-        this._drawNonLocatable = true;
-      });
 
-      dispose(fbo);
-    }
+    this.renderSystem.frameBufferStack.execute(resources.fbo, true, false, () => {
+      this._drawNonLocatable = !excludeNonLocatable;
+      result = this.readPixelsFromFbo(rect, selector);
+      this._drawNonLocatable = true;
+    });
 
-    dispose(texture);
+    this.disposeOrReuseReadPixelResources(resources);
 
     receiver(result);
 
     // Reset the batch IDs in all batches drawn for this call.
     this.uniforms.batch.resetBatchState();
   }
 
+  private createOrReuseReadPixelResources(rect: ViewRect): ReadPixelResources | undefined {
+    if (this._readPixelReusableResources !== undefined) {
+
+      // To reuse a texture, we need it to be the same size or smaller than what we need
+      if (this._readPixelReusableResources.texture.width >= rect.width && this._readPixelReusableResources.texture.height >= rect.height) {
+        const resources = this._readPixelReusableResources;
+        this._readPixelReusableResources = undefined;
+
+        return resources;
+      }
+    }
+
+    // Create a new texture/fbo
+    const texture = TextureHandle.createForAttachment(rect.width, rect.height, GL.Texture.Format.Rgba, GL.Texture.DataType.UnsignedByte);
+    if (texture === undefined)
+      return undefined;
+
+    const fbo = FrameBuffer.create([texture]);
+    if (fbo === undefined) {
+      dispose(texture);
+      return undefined;
+    }
+
+    return { texture, fbo };
+  }
+
+  private disposeOrReuseReadPixelResources({texture, fbo}: ReadPixelResources) {
+    if (this._readPixelReusableResources !== undefined) {
+      // Keep the biggest texture
```

Comment:
Occasionally we read a very large portion of the screen, or the entirety. For example, when the user turns the camera on, we read the depth of every pixel to determine the focus plane. Do we want to keep that big texture forever after that?

Reply:
Hmmm... Indeed. If so, that might be a good idea to avoid keeping always the largest.

Would limiting the size of the reused textures be a good solution? From what I have tested, most cases require textures smaller than 48x48 pixels, so a limit to 64px would work.

Reply:
The size of the pick rect for AccuSnap/auto-locate is controlled by ElementLocateManager.aperatureInches/touchApertureInches. NOTE: We read a larger area for touch since it's less precise. If this number needs to account for device pixels (hi def displays. 64px seems small)...

Reply:
Would it be better to attempt to compute this limit from the aperture value (Something like ``nextHighestPowerOfTwo(viewport.pixelsFromInches(elementLocateManager.touchApertureInches))``) or just set an higher limit like 256px? The ``Target`` class having no access to the ``ElementLocateManager``, the second option would be easier to implement.

Reply:
Probably just set a higher limit, @pmconne opinions?

Reply:
256 is more than large enough to accommodate our default pick aperture sizes (and any reasonable app-overridden larger sizes) and small enough not to worry about wasting memory. Sounds good to me.